docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -630,6 +630,150 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Users Module} \label{mUsers} 
+The Users Module exists to handle CRUD operations for operations related to users and their grocery trips.
+
+\subsection{Module}
+
+users
+
+\subsection{Uses}
+
+ Database Driver Module (Section \ref{mDBDriver}) 
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+\textbf{Item} as defined in Section \ref{mExtraction} \\
+
+\noindent \textbf{GroceryTrip} as defined in Section \ref{mExtraction} \\
+
+\noindent \textbf{User}
+
+\begin{table}[H]
+  \begin{tabular}{|p{0.25\textwidth}|p{0.25\textwidth}|p{0.40\textwidth}|}
+    \hline
+    \textbf{Output Name} & \textbf{Output Type} & \textbf{Description} \\
+    \hline
+    userId & Integer & Unique user id number \\
+    \hline
+    email & String & Email of user \\
+    \hline
+    password & String & Password of user (Encrypted) \\
+    \hline
+    firstName & String & First name of User \\
+    \hline
+    lastName & String & Last name of User \\
+    \hline
+    location & String & Location of user's home-base. (Coordinates) \\
+    \hline
+  \end{tabular}
+\end{table}
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{3cm} p{5cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+newUser & User & - & userExistsException \\
+getUser & String & User & userDneException \\
+updateUser & User & - & userDneException \\
+deleteUser & String & - & userDneException \\
+addTrip & GroceryTrip & - & userDneException \\
+getTrip & DateTime & GroceryTrip & - \\
```

Comment:
Inconsistency here in relation to the Analytics Module. There date time is treated as a natural number but here, it's of type DateTime.